### ISSUE TITLE
citizens/register

### ISSUE THREAD

**Issue by sleekLancelot** [BuildForSDGCohort2/TEAM-050-Backend#13]
In the register controller(/api/v1/citizen/register), after registration and you 'citizen.save()', can you make it return a token just like it does in the login controller? The way it is now it seems, one would have to login again after registering.

you can make it something like:
 ```
await citizen.save();
const payload = {
        id: citizen._id,
      };
      const token = jwt.sign(payload, mySecrete, { expiresIn: "1hr" });

      res.json({
        msg: "you are now logged in",
        token,
      });
```

**Comment by udofia2** [comment]
Thanks for raising the issue.
It is correct that after registration, the citizen will have to login.
The token is signed during login

**Comment by udofia2** [comment]
The will be form for registration.
After which the user will have to be redirected to the login page

**Comment by sleekLancelot** [comment]
> Thanks for raising the issue.
> It is correct that after registration, the citizen will have to login.
> The token is signed during login

yes, I was talking of also returning a signed token after registration, I might be mistaking but that is possible too right!?.
So the '/api/v1/citizen/profile/:id' controller can fetch the citizen and offence pertaining to the id in the header.
Registering and then having to subsequently login with the newly registered details might give poor mark on user-experience.

**Comment by sleekLancelot** [comment]
Also, when we hit 'GET /api/v1/citizen/profile/:id'
the id parameter in the req will course a lot of problems, its suppose to be a the citizen._id right?
but i can't get the citizen details until this code runs `const citizen = await Citizens.findOne({ _id: req.params.id });`
And that code only runs on a GET request to `/api/v1/citizen/profile/:id` which requires a param of the citizen id,whc i dont have yet

so i suggest you change the controller url from `/api/v1/citizen/profile/:id` to `/api/v1/citizen/profile`
And 
`const citizen = await Citizens.findOne({ _id: req.params.id });` 
 to something like :
`const citizen= await Citizens.findById(req.user, '-password');
        res.json({ citizen});`

**Comment by udofia2** [comment]
sorry for the delay.
And thank you for your patience
The issues are being attended to
